src/main/kotlin/org/dashj/platform/dpp/util/Cbor.kt

```diff
@@ -289,50 +193,45 @@ object Cbor {
             is Boolean -> arrayBuilder.add(value)
             is Int -> arrayBuilder.add(value.toLong())
             is BigInteger -> arrayBuilder.add(value.toLong())
-            is Number -> arrayBuilder.add(value.toLong())
             is Float -> arrayBuilder.add(value)
             is Long -> arrayBuilder.add(value)
             is Double -> arrayBuilder.add(value)
             is ByteArray -> arrayBuilder.add(value)
             is Identifier -> arrayBuilder.add(value.toBuffer())
+            is Number -> arrayBuilder.add(value.toLong())
             null -> arrayBuilder.add(SimpleValue(SimpleValueType.NULL))
-            else -> arrayBuilder.add(value.toString()) // ?
+            else -> throw IllegalArgumentException("No converter for $value")
```

Comment:
Now if an unknown type is included in a list or map, an exception will be thrown.  This will make it easier to find problems rather than seeing why a deep comparison failed.

`Identifier` with is a 32 byte array is rendered as a `ByteArray`.  The deep comparison functions are aware of this and will compare `Identifier`s with 32 byte arrays.  This is the only custom type included here.